source/MaterialXGenOsl/OslShaderGenerator.cpp

```diff
@@ -451,39 +466,43 @@ void OslShaderGenerator::emitShaderInputs(const VariableBlock& inputs, ShaderSta
         // Add shader input metadata.
         //
 
-        const std::unordered_map<const TypeDesc*, ShaderMetadata> UI_WIDGET_METADATA =
-        {
-            { Type::FLOAT, ShaderMetadata("widget", Type::STRING, Value::createValueFromStrings("number", Type::STRING->getName())) },
-            { Type::INTEGER, ShaderMetadata("widget", Type::STRING, Value::createValueFromStrings("number", Type::STRING->getName())) },
-            { Type::FILENAME, ShaderMetadata("widget", Type::STRING, Value::createValueFromStrings("filename", Type::STRING->getName())) },
-            { Type::BOOLEAN,  ShaderMetadata("widget", Type::STRING, Value::createValueFromStrings("checkBox", Type::STRING->getName())) }
-        };
-
         auto widgetMetadataIt = UI_WIDGET_METADATA.find(input->getType());
         const ShaderMetadata* widgetMetadata = widgetMetadataIt != UI_WIDGET_METADATA.end() ? &widgetMetadataIt->second : nullptr;
         const ShaderMetadataVecPtr& metadata = input->getMetadata();
+
         if (widgetMetadata || (metadata && metadata->size()))
         {
-            emitLineEnd(stage, false);
-            emitScopeBegin(stage, Syntax::DOUBLE_SQUARE_BRACKETS);
+            StringVec metadataLines;
             if (metadata)
             {
                 for (size_t j = 0; j < metadata->size(); ++j)
                 {
                     const ShaderMetadata& data = metadata->at(j);
-                    const string& delim = (widgetMetadata || j < metadata->size() - 1) ? Syntax::COMMA : EMPTY_STRING;
-                    const string& dataType = _syntax->getTypeName(data.type);
-                    const string dataValue = _syntax->getValue(data.type, *data.value, true);
-                    emitLine(dataType + " " + data.name + " = " + dataValue + delim, stage, false);
+                    if (METADATA_TYPE_BLACKLIST.count(data.type) == 0)
```

Comment:
This guard prevents generation of code like:
```
    vector2 uvtiling = {3, 3}
    [[
        vector2 min = {0, 0},      // Struct assignments in this context gives syntax error
        vector2 max = {25, 25}  //
    ]],
```

Instead the input will be generated without the metadata:
```
    vector2 uvtiling = {3, 3},
```